packages/jest-circus/src/legacy-code-todo-rewrite/jestAdapterInit.ts

```diff
@@ -122,7 +122,7 @@ export const initialize = ({
     });
 
   const {expand, updateSnapshot} = globalConfig;
-  const snapshotResolver = buildSnapshotResolver(config);
+  const snapshotResolver = buildSnapshotResolver(config, localRequire);
```

Comment:
This is not consistent with how snapshot serializers are handled (`localRequire()`d here and the actual modules passed into `jest-snapshot`), which I referred to in my suggestion on how to solve this.
Did you try doing it that way and hit some issues, or is there a particular reason why snapshot resolvers should be different?

Reply:
> This is not consistent with how snapshot serializers are handled

I had trouble figuring out how snapshot serializers were handled in the first place.

But the bigger factor was that doing that meant `buildSnapshotResolver` seemed to become redundant:

* `buildSnapshotResolver` checks the cache to see if the resolver has already been created,
* `createSnapshotResolver` checks if `snapshotResolverPath` is set, calling `createCustomSnapshotResolver` if it is otherwise `createDefaultSnapshotResolver`,
* `createCustomSnapshotResolver` checks that all the required properties exist on the required resolver, and that `verifyConsistentTransformations` doesn't throw.

Of the above, passing the `localRequire()`d resolver instead of `localRequire` would mean stipping out the cache (since it's for avoiding doing the same transformation multiple times), moving the `snapshotResolverPath` out, and refactoring `createCustomSnapshotResolver` to `verifyCustomSnapshotResolver`.

tbh I'd say the question here is actually why is there property validation for `snapshotResolver` but not `snapshotSerializers`?

Reply:
You're right, I only looked at the cache key which could still be used (because it's root dir, not snapshotResolverPath or something) but did not consider that the cache would be useless if the module is already required when entering the function.